packages/network-controller/src/rpc-service/rpc-service-chain.ts

```diff
@@ -99,11 +99,11 @@ export class RpcServiceChain implements RpcServiceRequestable {
    * @param fetchOptions - An options bag for {@link fetch} which further
    * specifies the request.
    * @returns The decoded JSON-RPC response from the endpoint.
-   * @throws A "method not found" error if the response status is 405.
-   * @throws A rate limiting error if the response HTTP status is 429.
-   * @throws A timeout error if the response HTTP status is 503 or 504.
-   * @throws A generic error if the response HTTP status is not 2xx but also not
-   * 405, 429, 503, or 504.
+   * @throws A 401 error if the response status is 401.
+   * @throws A "rate limiting" error if the response HTTP status is 429.
+   * @throws A "resource unavailable" error if the response status is 402, 404, or any 5xx.
```

Comment:
wondering why 404 doesn't get a 32002 instead

Reply:
Isn't that what "resource unavailable" is? https://github.com/MetaMask/rpc-errors/blob/4dd3f3dcedca8bbd57474915dbd92b892c548c5e/src/error-constants.ts#L5